### ISSUE TITLE
[Bug]: ignoring stable documentation link doesn't seem to work

### ISSUE THREAD

**Issue by lucaferranti** [JuliaBesties/BestieTemplate.jl#498]
### Description

The `.lychee.toml` file seems to exclude the link to the stable documentation from the link checks, in my case

```
  "^https://lucaferranti.github.io/DedekindCutArithmetic.jl/stable$",
```

which makes sense, since I may want to have the badge and link ready in the readme before starting the registration.

That doesn't quite seem to work tough:

See https://github.com/lucaferranti/DedekindCutArithmetic.jl/actions/runs/11475868486/job/31934649608

The only thing I'm doing differently from the generated template, is that I use markdown "named links"

i.e. I have

```markdown
[![Stable Documentation][stabledoc-img]][stabledoc-url]
```

for the badge and

```markdown
- [**STABLE**][stabledoc-url]: Documentation of the latest release
```

and at the bottom of the file

```markdown
[stabledoc-img]: https://img.shields.io/badge/docs-stable-blue.svg
[stabledoc-url]: https://lucaferranti.github.io/DedekindCutArithmetic.jl/stable/
```

maybe the regex cannot capture this usecase? :thinking: I am not familiar with lychee, but the `^` and `$` symbols generally match only if the content is the full line, hence the link would not be ignored if inlined? :thinking: 

### Package Version

0.13.0

### Julia Version

1.11

### Reproduction steps

Create a package with bestie template. In the readme, modify the links to stable version to use the format descibed in the description section above

### Operating System

Linux

**Comment by abelsiqueira** [comment]
We had a similar regex match issue with another link because of the `$` at the end. The `/` in your link is probably enough to break that - but I also haven't tried named links. We should fix the regex in the template `.lychee.toml`.

**Comment by lucaferranti** [comment]
I did do a bunch of copy-pasting from my own old readme template, it does seem indeed the `/` was introduced by that, since looking at another package generated with bestie, it's not there.

I haven't tested, but maybe the following could make it a bit more robust? (i.e. add `/?` at the end to make an extra slash optional)

```
  "^https://lucaferranti.github.io/DedekindCutArithmetic.jl/stable/?$"
```

One can still argue this is on me for changing the link generated by the template

**Comment by lucaferranti** [comment]
btw, is it intended that the link checker is not run with `pre-commit run -a` ?

**Comment by abelsiqueira** [comment]
The link regex should definitely be more robust. Even just removing the `$` from the original regex would work. I'm happy to accept a PR with either.

It is intended to not run with `pre-commit run -a`. We had another link-checker in the pre-commit, and it was too slow. lychee is faster, so maybe it makes sense to re-add.
It can be a new issue to add a new option to add lychee to the pre-commit, defaulting to false.
